apps/srt-tunnel.cpp

```diff
@@ -450,6 +450,40 @@ class TcpMedium: public Medium
 
 #endif
 
+#ifdef _WIN32
+    static int tcp_send(int socket, const char* buf, int bufsize)
+    {
+        return send(socket, buf, bufsize, 0);
+    }
+
+    static int tcp_recv(int socket, char* buf, int bufsize)
+    {
+        return recv(socket, buf, bufsize, 0);
+    }
+
+    static int tcp_close(int socket)
+    {
+        shutdown(socket, SD_BOTH);
+        return closesocket(socket);
+    }
+#else
+    static int tcp_send(int socket, const char* buf, int bufsize)
+    {
+        return ::write(socket, buf, bufsize);
+    }
+
+    static int tcp_recv(int socket, char* buf, int bufsize)
+    {
+        return ::read(socket, buf, bufsize);
+    }
+
+    static int tcp_close(int socket)
+    {
+        shutdown(socket, SD_BOTH);
```

Comment:
Doing `shutdown` before `close` is useless, as `close/closesocket` will still do this same thing first. If your intention was to wait until all data are in synch before closing, consider using `fsync`.